### PR TITLE
DOCSP-40801: Count documents

### PR DESCRIPTION
**Pull request by norareidy**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-40801
Staging - https://preview-mongodbnorareidy.gatsbyjs.io/cpp-driver/DOCSP-40801-count/read/count/

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [ ] Are all the links working?
- [ ] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/includes/read/count.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    // start-db-coll
+    auto db = client["sample_training"];
+    auto collection = db["companies"];
+    // end-db-coll
+ 
+    {
+        // Counts documents that have a "founded_year" value of 2010
+        // start-count-accurate
+        auto result = collection.count_documents(make_document(kvp("founded_year", 2010)));
+        std::cout << "Number of documents: " << result << std::endl;
```
Comment (rustagir) [inline source/includes/read/count.cpp:26]: S: give more descrriptive output
```suggestion
        std::cout << "Number of companies founded in 2010: " << result << std::endl;
```

File: source/read/count.txt [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+Count All Documents
+~~~~~~~~~~~~~~~~~~~
+
+To return a count of all documents in the collection, pass an empty filter document to
+the ``count_documents()`` method, as shown in the following example:
+
+.. code-block:: cpp
+
+   auto count = collection.count_documents({})
```
Comment (rustagir) [inline source/read/count.txt:65]: Q: why not include this in the code file as well? Also why not include a print statement?

File: source/read/count.txt [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+   .. input:: /includes/read/count.cpp
+      :start-after: start-modify-accurate
+      :end-before: end-modify-accurate
+      :language: cpp
+      :dedent:
+
+   .. output::
+
+      Number of documents: 100
```
Comment (rustagir) [inline source/read/count.txt:144]: S: make output more descriptive

**Review by rustagir — APPROVED** [review]
A few small things, doesnt need a second look unless you want!

File: source/read/count.txt (hunk 2 above)
Comment (norareidy) [reply source/read/count.txt]: moved to the code file & added output - it seemed like a basic syntax example, but I think it still benefits from those changes

File: source/read/count.txt [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+cluster and assign the following values to your ``db`` and ``collection`` variables:
+
+.. literalinclude:: /includes/read/count.cpp
+    :language: cpp
+    :dedent:
+    :start-after: start-db-coll
+    :end-before: end-db-coll
+
+To learn how to create a free MongoDB Atlas cluster and load the sample datasets, see the
+:ref:`<cpp-get-started>` tutorial.
```
Comment (eramongodb) [inline source/read/count.txt:44]: Is this reference correct? I do not think the "Get Started with the C++ Driver" pages are relevant to creating a free MongoDB Atlas cluster.

File: source/includes/read/count.cpp [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        std::cout << "Number of companies founded in 2010: " << result << std::endl;
+        // end-count-accurate
+    }
+
+    {
+        // Counts a maximum of 100 documents that have a "number_of_employees" value of 50
+        // start-modify-accurate
+        mongocxx::options::count opts;
+        opts.limit(100); 
+        auto result = collection.count_documents(make_document(kvp("number_of_employees", 50)));
```
Comment (eramongodb) [inline source/includes/read/count.cpp:42]:
```suggestion
        auto result = collection.count_documents(make_document(kvp("number_of_employees", 50)), opts);
```
Missing `opts` argument specifying the limit set above.

**Review by eramongodb — CHANGES_REQUESTED** [review]

File: source/read/count.txt (hunk 4 above)
Comment (norareidy) [reply source/read/count.txt:44]: The Get Started tutorial includes a page ([Create a MongoDB Deployment](https://preview-mongodbnorareidy.gatsbyjs.io/cpp-driver/DOCSP-40801-count/get-started/create-a-deployment/)) that walks you through how to create an Atlas cluster. However it does link out to the Atlas docs for details, so I can link directly to the Atlas page here instead if you'd prefer

**Review by eramongodb — APPROVED** [review]
Pending suggestion regarding updating link to Atlas cluster setup guide; otherwise, LGTM.

File: source/read/count.txt (hunk 4 above)
Comment (norareidy) [reply source/read/count.txt]: Ok! I think it makes sense to link to the Atlas page; updated

**Pull request merged**